wiki_doc

```diff
@@ -0,0 +1,138 @@
+## Valkey Commands
+
+For information on the supported commands and their corresponding parameters, we recommend referring to the documentation in the code. This documentation provides in-depth insights into the usage and options available for each command.
+
+### Command String Arguments
+
+Valkey strings store sequences of bytes, that may include text, serialized objects, or binary arrays. As such, to pass Valkey strings as arguments to commands, or receive Valkey strings in responses, Glide offers two APIs:
+
+1. string: a valid UTF-8 string.
+2. GlideString: a union type of either a valid UTF-8 string or array of bytes.
+
+The return types of command API:
+
+-   Command uses `string` as the return type if it constains a specific string value. e.g. `public async type(key: GlideString): Promise<string>`
+-   Command uses `Buffer` as the return type if it contains binary data . e.g. `public async dump(key: GlideString): Promise<Buffer | null>`
+-   Most commands use `GlideString` as the common return type.
+
+### Decoder
+
+In Node.js, most users expect the responses to be in string format unless specified otherwise. Users can select which data type they expect to receive as a response. The user has the option of converting the response into either a string or bytes for each command. In addition, as part of the client creation process, the user has the option of setting a default for all commands if no decoder argument is passed to the command.
+
+There are two optional `Decoder` enum:
+
+1. `Bytes`: decodes the response into a buffer array.
+2. `String`: decodes the response into a string.
+
+#### Example Usage
+
+Here is the command implementation to outline arguments and response type:
+
+```typescript
+public getdel(
+        key: GlideString,
+        decoder?: Decoder,
+    ): Promise<GlideString | null>
+```
+
```

Comment:
Tell about client-scope default decoder, show how it could be configured.
Tell that if no decoder given nor configured, string decoder is used.
Warn user that data could be lost in some cases (see #2221)